src/Microsoft.Health.Fhir.Core/Features/Search/Registry/SearchParameterStatusManager.cs

```diff
@@ -88,63 +88,43 @@ public async Task EnsureInitialized()
-                        {
-                            if (resourceTypeSearchParamStatusMap.ContainsKey(resourceType))
-                            {
-                                resourceTypeSearchParamStatusMap[resourceType].Add(result);
-                            }
-                            else
-                            {
-                                resourceTypeSearchParamStatusMap.Add(resourceType, new List<ResourceSearchParameterStatus>() { result });
-                            }
-                        }
-                    }
+            await _mediator.Publish(new SearchParametersUpdated(updated));
+        }
 
-                    if (p.BaseResourceTypes != null)
-                    {
-                        foreach (string resourceType in p.BaseResourceTypes)
-                        {
-                            if (resourceTypeSearchParamStatusMap.ContainsKey(resourceType))
-                            {
-                                resourceTypeSearchParamStatusMap[resourceType].Add(result);
-                            }
-                            else
-                            {
-                                resourceTypeSearchParamStatusMap.Add(resourceType, new List<ResourceSearchParameterStatus>() { result });
-                            }
-                        }
-                    }
-                }
-            }
+        public async Task UpdateSearchParameterStatusAsync(IReadOnlyCollection<string> searchParameterUris, SearchParameterStatus status)
+        {
+            var searchParameterStatusList = new List<ResourceSearchParameterStatus>();
+            var updated = new List<SearchParameterInfo>();
 
-            var resourceHashMap = new Dictionary<string, string>();
-            foreach (KeyValuePair<string, List<ResourceSearchParameterStatus>> kvp in resourceTypeSearchParamStatusMap)
+            foreach (string uri in searchParameterUris)
             {
-                string searchParamHash = SearchHelperUtilities.CalculateSearchParameterHash(kvp.Value);
-                resourceHashMap.Add(kvp.Key, searchParamHash);
+                var searchParamUri = new Uri(uri);
+
+                var paramInfo = _searchParameterDefinitionManager.GetSearchParameter(searchParamUri);
+                updated.Add(paramInfo);
+                paramInfo.IsSearchable = status == SearchParameterStatus.Enabled;
+                paramInfo.IsSupported = status != SearchParameterStatus.Disabled;
+
+                searchParameterStatusList.Add(new ResourceSearchParameterStatus()
+                {
+                    LastUpdated = Clock.UtcNow,
+                    Status = status,
+                    Uri = searchParamUri,
+                });
             }
 
-            await _mediator.Publish(new SearchParametersHashUpdated(resourceHashMap));
+            await _searchParameterStatusDataStore.UpsertStatuses(searchParameterStatusList);
```

Comment:
We had a bug in the beginning with multiple instances over-writing each other, I haven't traced the calls to this, just FYI